src/Core/ViewHelper/ViewHelperVariableContainer.php

```diff
@@ -29,6 +30,53 @@ class ViewHelperVariableContainer
      */
     protected $view;
 
+    /**
+     * Push a new delegate variable container to a stack. Done before you start
+     * sub-rendering with a ViewHelper that requires a special container for
+     * variables which is capable of being used in multidepth stacks without
+     * overriding the last instance or last set of variables.
+     *
+     * @param string $viewHelperClassName
+     * @param VariableProviderInterface $variableProvider
+     */
+    public function pushDelegateVariableContainer($viewHelperClassName, VariableProviderInterface $variableProvider)
```

Comment:
The term `overlay` would be better than `delegate` IMO.

Reply:
I both agree and disagree: in this specific use case you are right that we are dealing with a sort of `overlay` logic so semantics would imply that naming. But, the methods are actually written to allow a similar approach to be used for other use cases as well - including but not limited to `f:switch` which currently has a problem with nested usages because it doesn't use a stack.

Migrating `f:switch` to this pattern would make the `overlay` naming semantically inappropriate - the next best thing is a "stack of delegates" that can receive variables but aren't the main source of variables for template rendering (is used as transient stack-based storage).

Naming is hard :)